src/lib/vm.ts

```diff
@@ -0,0 +1,95 @@
+import type { Command } from "@commander-js/extra-typings";
+import { isFeatureEnabled } from "./posthog.ts";
+import { apiClient } from "../apiClient.ts";
+import { readFileSync } from "node:fs";
+import {
+	logAndQuit,
+	logSessionTokenExpiredAndQuit,
+} from "../helpers/errors.ts";
+
+export async function registerVM(program: Command) {
+	const isEnabled = await isFeatureEnabled("vms");
+
+	if (!isEnabled) {
+		return;
+	}
+
+	const vm = program
+		.command("vm")
+		.aliases(["v", "vms"])
+		.description("Manage virtual machines");
+
+	const client = await apiClient();
+
+	vm.command("list")
+		.description("List all virtual machines")
+		.action(async () => {
+			const { data, error, response } = await client.GET("/v0/vm/nodes");
+
+			if (!response.ok) {
+				if (response.status === 401) {
+					await logSessionTokenExpiredAndQuit();
+				}
+				logAndQuit(`Failed to list VMs: ${error?.message}`);
+			}
+
+			if (!data?.data) {
+				logAndQuit("No VMs found");
+			}
+
+			console.table(
+				data.data.map((node) => ({
+					id: node.id,
+					status: node.status,
+					created_at: node.created_at,
+				})),
+			);
+		});
+
+	vm.command("script")
+		.description("Push a startup script to VMs")
+		.requiredOption("-f, --file <file>", "Path to startup script file")
+		.action(async (options) => {
+			let script: string;
+			try {
+				script = readFileSync(options.file, "utf-8");
+			} catch (err) {
+				logAndQuit(`Failed to read script file: ${err.message}`);
+			}
```

Comment:
Fun fact JS used to have a `with` keyword

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/with

Not supported in strict JS/TS though

Reply:
DO NOT DO THIS.